internal/syscallfs/readfs_test.go

```diff
@@ -11,22 +11,50 @@ import (
 	"github.com/tetratelabs/wazero/internal/testing/require"
 )
 
-func TestReadFS_MkDir(t *testing.T) {
+func TestNewReadFS(t *testing.T) {
 	tmpDir := t.TempDir()
-	testFS := NewReadFS(Adapt("/", hackFS(tmpDir)))
 
-	err := testFS.Mkdir("mkdir", fs.ModeDir)
+	// Doesn't double-wrap file systems that are already read-only
+	adapted := Adapt(os.DirFS(tmpDir), "/")
+	require.Equal(t, adapted, NewReadFS(adapted))
+	require.Equal(t, EmptyFS, NewReadFS(EmptyFS))
+
+	// Wraps a writeable file system
+	writeable, err := NewDirFS(tmpDir, "/tmp")
+	require.NoError(t, err)
+	readFS := NewReadFS(writeable)
+	require.NotEqual(t, writeable, readFS)
+	require.Equal(t, writeable.GuestDir(), readFS.GuestDir())
+}
+
+func TestReadFS_String(t *testing.T) {
+	writeable, err := NewDirFS(".", "/tmp")
+	require.NoError(t, err)
+
+	readFS := NewReadFS(writeable)
+	require.NotEqual(t, writeable, readFS)
+	require.Equal(t, "./:/tmp:ro", readFS.String())
```

Comment:
This is a lot better in error messages as it matches our CLI syntax (like docker volume)